src/cocoa_init.m

```diff
@@ -372,3 +371,43 @@ void _glfwPlatformTerminate(void)
         ;
 }
 
+static BOOL isNSApplicationRunning()
+{
+    BOOL isRunning = NO;
+    NSApplication *sharedNSApp = [NSApplication sharedApplication];
+    if(sharedNSApp)
```

Comment:
I think you're wrong: when the variable is declared in the if statement, it is available in the if block. Please try again.

Reply:
Or maybe you're right and I'm assuming it's C++, but it's C and you can't do that in C?
Anyway it's not a big deal.

Reply:
I think that's a C++ feature, and not valid in C. Note that these are Objective-C source files, which inherit their syntax rules from C.

Reply:
@tombsar Yes, and I'm fine with leaving it as-is. Juts out of curiosity, do you think turning the file to a .mm file enable C++ syntax rules? I remember that XCode interprets .mm files as C++ + Objective C.

Reply:
You can have objective-C++ (which is .mm), but I think this is a bad idea because C++11 is by far the most complicated computer language ever created, and then you combine C++11 with objective-C and you increase double the complexity again. 

I think leaving it as just plain objective-c, with the relatively simple C/objective-C syntax / semantics is the by far the simplest and easiest to understand solution. 

I think it is much better to keep things as simple and as easy to understand as possible.

Reply:
Again, I'm fine with leaving it as-is, so no problem here.

Yet, I must say, as a strong believer in modern C++, I disagree "strongly" :) with your statement regarding c++11. We can discuss that somewhere else if you want to not pollute this thread too much :)